AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/Support/MutationEventExtensionsTests.swift

```diff
@@ -0,0 +1,95 @@
+        let updateMutationEvent = MutationEvent(id: UUID().uuidString,
+                                                        modelId: modelId,
+                                                        modelName: Post.modelName,
+                                                        json: "",
+                                                        mutationType: .update,
+                                                        version: nil)
+        let deleteMutationEvent = MutationEvent(id: UUID().uuidString,
+                                                        modelId: modelId,
+                                                        modelName: Post.modelName,
+                                                        json: "",
+                                                        mutationType: .delete,
+                                                        version: nil)
+        let metadata = MutationSyncMetadata(id: modelId,
+                                            deleted: false,
+                                            lastChangedAt: Int(Date().timeIntervalSince1970),
+                                            version: 1)
+
+        let createMutationSync = MutationSync(model: post, syncMetadata: metadata)
+
+        let updateMutationExpectation = expectation(description: "save updateMutationEvent success")
+        storageAdapter.save(updateMutationEvent) { result in
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            updateMutationExpectation.fulfill()
+        }
+        wait(for: [updateMutationExpectation], timeout: 1)
+
+        let deleteMutationExpectation = expectation(description: "save deleteMutationEvent success")
+        storageAdapter.save(deleteMutationEvent) { result in
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            deleteMutationExpectation.fulfill()
+        }
+        wait(for: [deleteMutationExpectation], timeout: 1)
+
+        let queryAfterUpdatingVersionExpectation = expectation(description: "query success")
+        let updatingVersionExpectation = expectation(description: "update version success")
+        MutationEvent.updatePendingMutationEventVersionIfNil(for: post.id,
+                                                        mutationSync: createMutationSync,
+                                                        storageAdapter: storageAdapter) { result in
+            switch result {
+            case .failure(let error):
+                XCTFail("Error : \(error)")
+            case .success:
+                // query for head of mutation event table for given model id and check for version
+                updatingVersionExpectation.fulfill()
+                MutationEvent.pendingMutationEvents(for: post.id,
```

Comment:
what is the point of this expectation if you will perform something after it? should be like:
```
MutationEvent.updatePendingMutationEventVersionIfNil  {
  updatingVersionExpectation.fulfill() // unblock your wait
}
wait(updatingVersionExpectation) // wait for your action to complete
MutationEvent.pendingMutationEvents // continue your test
```